TemplateApplication/Resources/SocialSupportQuestionnaire.json

```diff
@@ -360,6 +293,95 @@
           }
         }
       ]
+    },
+    {
+      "linkId": "c3bea33d-4c50-4f4a-8ae4-1a52be326b19",
+      "type": "string",
+      "text": "What is your phone number? Ex. (555) 555-5555",
+      "required": false,
+      "enableWhen": [
+        {
+          "question": "695525f3-3e89-4455-8e25-878171c596da",
+          "operator": "=",
+          "answerCoding": {
+            "system": "urn:uuid:736ac230-812a-4f4a-edec-5156910fb6ec",
+            "code": "phone-call"
+          }
+        }
+      ],
+      "extension": [
+        {
+          "url": "http://hl7.org/fhir/StructureDefinition/regex",
+          "valueString": "^(\\([0-9]{3}\\) |[0-9]{3}-)[0-9]{3}-[0-9]{4}$"
+        },
+        {
+          "url": "http://biodesign.stanford.edu/fhir/StructureDefinition/validationtext",
+          "valueString": "Please enter a valid phone number."
+        }
+      ]
+    },
+    {
+      "linkId": "8e906a39-5fd0-42a8-f42c-bd96d719dd13",
+      "type": "string",
+      "text": "What is your text number? Ex. (555) 555-5555",
+      "required": false,
+      "enableWhen": [
+        {
+          "question": "695525f3-3e89-4455-8e25-878171c596da",
+          "operator": "=",
+          "answerCoding": {
+            "system": "urn:uuid:736ac230-812a-4f4a-edec-5156910fb6ec",
+            "code": "text-message"
+          }
+        }
+      ],
+      "extension": [
+        {
+          "url": "http://hl7.org/fhir/StructureDefinition/regex",
+          "valueString": "^(\\([0-9]{3}\\) |[0-9]{3}-)[0-9]{3}-[0-9]{4}$"
+        },
+        {
+          "url": "http://biodesign.stanford.edu/fhir/StructureDefinition/validationtext",
+          "valueString": "Please enter a valid phone number."
+        }
+      ]
+    },
+    {
+      "linkId": "86290b0a-017e-4193-8707-dc0c2146f0eb",
+      "type": "string",
+      "text": "What is your e-mail?",
+      "extension": [
+        {
+          "url": "http://hl7.org/fhir/StructureDefinition/regex",
+          "valueString": "^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$"
```

Comment:
Do we have guidelines on how we want to handle E-Mail validation and verification?

We currently have a mixed set of E-Mail regular expression around all Spezi modules (e.g. SpeziAccount maintains one as well). All regex I encountered so far don't specify their source and also match against a grammar that is diverging from [RF5322](https://datatracker.ietf.org/doc/html/rfc5322#section-3.4.1). While the above regex seems fine in that sense that it isn't too restrictive (it still captures less that what is allowed with the RFC afaik), it still wanted to use this opportunity to start this discussion.

There are generally two concerns:
* We cause adoption of email Regular Expressions within the framework ecosystem and dependent projects that may reject perfectly valid E-Mail addresses.
* Having a RFC compliant or "properly" formatted email address doesn't give us any information about its existence. Ultimately, an email may only considered valid if the user can proof being able to receive emails on that address.

So there are really two questions to this discussion:
* What type of Regex do we agree on to use within the Spezi ecosystem (maybe even use-case-dependent?). As an example, in the upcoming SpeziAccount release I'm currently striving for a more minimal validation as it is backed by E-Mail validation of the Account Service anyways (see [here](https://github.com/StanfordSpezi/SpeziAccount/blob/297fc05e57ed568ca2a7a3b0c284e1e02c6f05cb/Sources/SpeziAccount/Model/Validation/ValidationRule%2BDefaults.swift#L48)). There are great StackOverflow discussions providing more insights about [parsing](https://stackoverflow.com/a/48170419) and the [general problem domain](https://stackoverflow.com/a/201378).
* Validating E-Mails (and phone numbers) is a problem other people using the framework might likewise encounter. Do we have the ability to rely on sending E-Mail verifications for data provided in questionnaires? Could we provide a way to source that data from other components (e.g. the currently logged in user from SpeziAccount)?

Might make sense to move this discussion somewhere else. This should not block your PR.

Reply:
Hi @PSchmiedmayer, it is [defined in Phoenix]( https://github.com/StanfordBDHG/phoenix/blob/4bfefd80fcacd0e4f46666708f5eb0ba98d69e24/src/components/Question/ValidationAnswerTypes/ValidationAnswerTypeString.tsx#L25). We can change it. There is no specific regex defined in the FHIR standard to my knowledge.

Reply:
Sounds great, let's change that to the very simple regex that @Supereg is using [here]( https://github.com/StanfordSpezi/SpeziAccount/blob/297fc05e57ed568ca2a7a3b0c284e1e02c6f05cb/Sources/SpeziAccount/Model/Validation/ValidationRule%2BDefaults.swift#L48).

Different discussion, but I think we should generally also think about the other RegEx expressions in Phoenix, e.g. the Zip code is US-specific (maybe just renaming it to US-Zip is the way to go) and the URL Regex doesn't seem to match, e.g. new top level domains that are longer than 6 characters are not matched and the focus on `www` and `ftp` might be too specific for the URL. But that might be a discussion over in Phoenix 🚀

Let's just update the RegEx here and merge the PR 🥳